lib/programmemory.cpp

```diff
@@ -1411,18 +1411,34 @@ namespace {
 
         ValueFlow::Value executeImpl(const Token* expr)
         {
-            const ValueFlow::Value* value = nullptr;
             if (!expr)
                 return unknown();
-            if (expr->hasKnownIntValue() && !expr->isAssignmentOp() && expr->str() != ",")
-                return expr->values().front();
-            if ((value = expr->getKnownValue(ValueFlow::Value::ValueType::FLOAT)) ||
-                (value = expr->getKnownValue(ValueFlow::Value::ValueType::TOK)) ||
-                (value = expr->getKnownValue(ValueFlow::Value::ValueType::ITERATOR_START)) ||
-                (value = expr->getKnownValue(ValueFlow::Value::ValueType::ITERATOR_END)) ||
-                (value = expr->getKnownValue(ValueFlow::Value::ValueType::CONTAINER_SIZE))) {
-                return *value;
+            const ValueFlow::Value* value = nullptr;
+            for (const auto& val : expr->values())
+            {
+                if (!val.isKnown())
+                    continue;
+                switch (val.valueType) {
+                    case ValueFlow::Value::ValueType::INT: {
+                        if (!expr->isAssignmentOp() && expr->str() != ",")
+                            return expr->values().front();
+                        break;
+                    }
+                    case ValueFlow::Value::ValueType::FLOAT:
+                    case ValueFlow::Value::ValueType::TOK:
+                    case ValueFlow::Value::ValueType::ITERATOR_START:
+                    case ValueFlow::Value::ValueType::ITERATOR_END:
+                    case ValueFlow::Value::ValueType::CONTAINER_SIZE: {
+                        if (!value)
+                            value = &val;
+                        break;
+                    }
+                    default:
+                        break;
+                }
```

Comment:
I think it would be better to make this a function that returns a pointer to the value. Then we can just write `if((const ValueFlow::Value* value = findKnownValue(expr->values())))` here.

Reply:
Makes sense.

But then we would need to exclude the `INT` handling since that currently has priority and a different handling.